internal/app/networkd/pkg/nic/nic.go

```diff
@@ -273,9 +271,15 @@ func (n *NetworkInterface) configureInterface(method address.Addressing) error {
 
 	// Add any routes
 	for _, r := range method.Routes() {
+		// If gateway/router is 0.0.0.0 we'll set to nil so route scope decision will be correct
+		gw := r.Router
+		if net.IPv4zero.Equal(gw) {
+			gw = nil
+		}
+
 		// Any errors here would be non-fatal, so we'll just ignore them
-		// nolint: errcheck
-		n.rtnlConn.RouteAddSrc(method.Link(), *r.Dest, method.Address(), r.Router)
+		//nolint: errcheck
+		n.rtnlConn.RouteAddSrc(method.Link(), *r.Dest, method.Address(), gw)
```

Comment:
should we log the errors in case it helps with the debugging? or that is something we don't really need?

Reply:
once we get log levels implemented i'd very much like to spit out errors